fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaData.java

```diff
@@ -99,6 +99,25 @@ public class RecordMetaData implements RecordMetaDataProvider {
             RecordMetaDataProto.getDescriptor(), RecordMetaDataOptionsProto.getDescriptor(), TupleFieldsProto.getDescriptor()
     };
 
+    protected RecordMetaData(@Nonnull RecordMetaData orig) {
```

Comment:
This doesn't seem to be used in this branch, but I wonder whether a `toBuilder` / `asBuilder` would be more idiomatic.

Reply:
This gets used in the consumer, where a copy-constructor is used to create an isolated instance from the original.
I thought of using a `toBuilder` pattern but ran into difficulties as the building of `RecordMetaDataBuilder` from `RecordMetadaData` is not trivial: For example, the reverse engineering of a a `RecordTypeBuilder` and the Indexes seemed too involved. I would be happy to do so if there is a path though.

Reply:
I suppose it can wait, then.